File: ai-workspace-web/src/api/mcp.ts
/**
 * MCP 服务器注册表 API
 *
 * MCP 服务器按 createBy 用户私有，支持 sse/stdio 两种传输方式。
 * testMcpServer 仅对 sse 类型做 HTTP 可达性探测，返回 { reachable, latency, message }。
 * Agent 运行时会通过 langchain-mcp-adapters 动态加载 SSE 类型服务器的工具列表。
 */
import request from './request'

export interface McpServer {
  id?: number
  name: string
  description?: string
  transport?: string
  url?: string
  command?: string
  config?: string
  enabled?: number
  createTime?: string
}

export const listMcpServers = () => request.get<unknown, McpServer[]>('/mcp/list')
export const addMcpServer = (data: McpServer) => request.post('/mcp/add', data)
export const updateMcpServer = (data: McpServer) => request.put('/mcp/update', data)
export const deleteMcpServer = (id: number) => request.delete(`/mcp/delete/${id}`)
export interface McpTestResult {
  reachable: boolean
  latency: number
  message: string
}

export const testMcpServer = (id: number) =>
  request.post<unknown, McpTestResult>(`/mcp/test/${id}`)

File: ai-workspace-backend/internal/model/mcp.go
package model

// McpServer MCP 服务器注册信息，对应 mcp_server 表。
// 用户私有；sse 类型可在 Agent 运行时通过 langchain-mcp-adapters 远程加载工具；
// stdio 类型仅本地可用，Agent 运行时不支持。
type McpServer struct {
	UserOwnedModel
	Name        string `gorm:"column:name;size:100"         json:"name"`
	Description string `gorm:"column:description;size:500"  json:"description"`
	// sse: HTTP SSE 传输；stdio: 标准输入输出（仅本地）
	Transport   string `gorm:"column:transport;size:20"     json:"transport"`
	// SSE 服务器基础 URL，连通性检测与 Agent 运行时均使用此字段
	URL         string `gorm:"column:url;size:500"          json:"url"`
	// stdio 传输的命令行启动指令
	Command     string `gorm:"column:command;size:500"      json:"command"`
	// 扩展配置 JSON：{"headers":{...},"env":{...},"args":[...]}
	Config      string `gorm:"column:config;type:text"      json:"config"`
	// 1=启用，0=禁用
	Enabled     int8   `gorm:"column:enabled;default:1"     json:"enabled"`
}

func (McpServer) TableName() string { return "mcp_server" }
